### DISCUSSION TITLE
How to set parameter names with `uamethod`

### DISCUSSION THREAD

**Discussion by danmoser** [FreeOpcUa/opcua-asyncio#852] [Q&A] [answered]
Hi folks,
Does anyone know how to set names for the input and output parameters in the example below?
Thanks!

```
    @uamethod
    def defined_function(parent, bvar):
        if bvar:
            return "True"
        return "False"

    ...

    await my_obj.add_method(
        idx,
        "myFunction",
        defined_function,
        [ua.VariantType.Boolean],
        [ua.VariantType.String],
    )
```

**Comment by AndreasHeine** [comment, marked as answer]
return a tuple of ua.Variant()? order matters!
https://github.com/FreeOpcUa/opcua-asyncio/blob/eabd784f05f2bbb8c177947c503d18d874982638/examples/server-extension-object-as-method-argument.py#L18

hint:
you should not do -> [ua.VariantType.Boolean] -> use ua.Argument()
https://github.com/FreeOpcUa/opcua-asyncio/blob/eabd784f05f2bbb8c177947c503d18d874982638/examples/server-extension-object-as-method-argument.py#L35

**Comment by danmoser** [comment]
Thank you, @AndreasHeine !
